dbms/src/TestUtils/mockExecutor.cpp

```diff
@@ -274,52 +276,112 @@ DAGRequestBuilder & DAGRequestBuilder::buildAggregation(ASTPtr agg_funcs, ASTPtr
 }
 
-void MockDAGRequestContext::addExchangeRelationSchema(String name, const MockColumnInfoList & columns)
+void MockDAGRequestContext::addExchangeRelationSchema(String name, const MockColumnInfoList & columnInfos)
 {
-    std::vector<MockColumnInfo> v_column_info(columns.size());
+    std::vector<MockColumnInfo> v_column_info(columnInfos.size());
     size_t i = 0;
-    for (const auto & info : columns)
+    for (const auto & info : columnInfos)
     {
         v_column_info[i++] = std::move(info);
     }
     exchange_schemas[name] = v_column_info;
 }
 
+void MockDAGRequestContext::addMockTableColumnData(const String & db, const String & table, ColumnsWithTypeAndName columns)
+{
+    mock_table_columns[db + "." + table] = columns;
+}
+
+void MockDAGRequestContext::addMockTableColumnData(const MockTableName & name, ColumnsWithTypeAndName columns)
+{
+    mock_table_columns[name.first + "." + name.second] = columns;
+}
+
+void MockDAGRequestContext::addExchangeReceiverColumnData(const String & name, ColumnsWithTypeAndName columns)
+{
+    mock_exchange_columns[name] = columns;
+}
+
+void MockDAGRequestContext::addMockTable(const String & db, const String & table, const MockColumnInfoList & columnInfos, ColumnsWithTypeAndName columns)
+{
+    addMockTable(db, table, columnInfos);
+    addMockTableColumnData(db, table, columns);
+}
+
+void MockDAGRequestContext::addMockTable(const String & db, const String & table, const MockColumnInfos & columnInfos, ColumnsWithTypeAndName columns)
+{
+    addMockTable(db, table, columnInfos);
+    addMockTableColumnData(db, table, columns);
+}
+
+void MockDAGRequestContext::addMockTable(const MockTableName & name, const MockColumnInfoList & columnInfos, ColumnsWithTypeAndName columns)
+{
+    addMockTable(name, columnInfos);
+    addMockTableColumnData(name, columns);
+}
+
+void MockDAGRequestContext::addMockTable(const MockTableName & name, const MockColumnInfos & columnInfos, ColumnsWithTypeAndName columns)
+{
+    addMockTable(name, columnInfos);
+    addMockTableColumnData(name, columns);
+}
+
+void MockDAGRequestContext::addExchangeReceiver(const String & name, MockColumnInfos columnInfos, ColumnsWithTypeAndName columns)
+{
+    addExchangeRelationSchema(name, columnInfos);
+    addExchangeReceiverColumnData(name, columns);
+}
+
+void MockDAGRequestContext::addExchangeReceiver(const String & name, MockColumnInfoList columnInfos, ColumnsWithTypeAndName columns)
+{
+    addExchangeRelationSchema(name, columnInfos);
+    addExchangeReceiverColumnData(name, columns);
+}
+
 DAGRequestBuilder MockDAGRequestContext::scan(String db_name, String table_name)
 {
-    return DAGRequestBuilder(index).mockTable({db_name, table_name}, mock_tables[db_name + "." + table_name]);
+    auto builder = DAGRequestBuilder(index).mockTable({db_name, table_name}, mock_tables[db_name + "." + table_name]);
+    if (mock_table_columns.find(db_name + "." + table_name) != mock_table_columns.end())
```

Comment:
It will do nothing if not find  the table name?

Reply:
Yes. In this case,  user must put input columns in executeStreams in order to run the query.
Maybe I need to comment it out.

Reply:
ok.